### ISSUE TITLE
[BUG] Standard collections as type hints are not subscriptable (PEP 585)

### ISSUE THREAD

**Issue by d0tcc** [cython/cython#4969]
**Describe the bug**
When using the standard collections dict or list as a type hint in Python 3.10 ([PEP 585](https://peps.python.org/pep-0585/) released in Python 3.9) the cythonized code breaks with `TypeError: 'type' object is not subscriptable`

**To Reproduce**
Code to reproduce the behaviour:
```python
def do_something(self, foo: list[tuple[str, float]], bar: dict[str, float]) -> list[tuple[str, float]]:
  return foo
```

**Expected behavior**

**Environment (please complete the following information):**
 - OS: Linux
 - Python version 3.10.4
 - Cython version 0.29.32

**Comment by scoder** [comment]
Works for me in Cython 3.0. Will not be backported to the legacy stable 0.29.x series.

**Comment by da-woods** [comment]
It's to do with `__class_getitem__` which is a Cython 3 only feature. The other difference with Cython 3 is that annotations are converted to strings rather than evaluated (so it probably doesn't even use `__class_getitem__`).

Note that Cython 3 doesn't currently understand subscripted `list[...]`, `dict[...]` etc as type annotations. It understands `typing.List[...]` but not the plain types.

**Comment by yudonghai** [comment]
> Works for me in Cython 3.0. Will not be backported to the legacy stable 0.29.x series.

@scoder It seems that Python 3.0 has not yet released? The latest build is 0.29.34
